app/models/manageiq/providers/google/cloud_manager/provision/disk.rb

```diff
@@ -13,6 +13,13 @@ def create_disk(disk_attrs)
     create_disks([disk_attrs]).first
   end
 
+  def delete_disk(disk_attrs)
+    source.with_provider_connection do |google|
+      disk = google.disks.get(disk_attrs[:name], disk_attrs[:zone_name])
+      disk.destroy
```

Comment:
Is there a chance that nothing will be returned

Reply:
:+1: I thought this raised an exception if not found but it looks like it does return `nil` good call